### ISSUE TITLE
Clang-tidy: False positive readability-redundant-inline-specifier on out-of-line explicitly defaulted constructors

### ISSUE THREAD

**Issue by Eisenwave** [llvm/llvm-project#130745]
https://godbolt.org/z/Eqf1Er9M5
```cpp
struct awoo {
    ~awoo();
};

inline awoo::~awoo() = default;
```
```
<source>:5:1: warning: function '~awoo' has inline specifier but is implicitly inlined [readability-redundant-inline-specifier]
    5 | inline awoo::~awoo() = default;
      | ^~~~~~
1 warning generated.
```
This warning is wrong: https://eel.is/c++draft/dcl.fct.def.default#3

The out-of-line explicitly-defaulted definition is not the first declaration, so it is not implicitly `inline`.

**Comment by firewave** [comment]
Also supported by CppInsights: https://cppinsights.io/s/3a4ee4b9.
